packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -527,30 +529,9 @@ const ApplicationShell = <AdditionalEnvironmentProperties extends {}>(
           if (isAuthenticated)
             return (
               <Switch>
-                {/* When the application redirects to this route,
-                we always force a hard redirect to the logout route of
-                the authentication service. */}
-                <Route
-                  path="/logout"
-                  render={({ location }) => (
-                    <Redirector
-                      to="logout"
-                      location={location}
-                      queryParams={{
-                        reason: LOGOUT_REASONS.USER,
-                        ...(coercedEnvironmentValues.servedByProxy
-                          ? {}
-                          : {
-                              // This will be used after being logged in, to redirect to this location.
-                              redirectTo: window.location.origin,
-                            }),
-                      }}
-                    />
-                  )}
-                />
+                <Route path="/logout" component={RedirectToLogout} />
```

Comment:
The "logout redirector" has been moved to a component, so that I can use the hook to get the env from the application context